webhook/models/webhook.py

```diff
@@ -0,0 +1,205 @@
+        'Consumer name',
+        required=True,
+        help='Name of your consumer webhook. '
+             'This name will be used in named of event methods')
+    address_ids = fields.One2many(
+        'webhook.address', 'webhook_id', 'IP or Network Address',
+        required=True,
+        help='This address will be filter to know who is '
+             'consumer webhook')
+    python_code_get_event = fields.Text(
+        'Get event',
+        required=True,
+        help='Python code to get event from request data.\n'
+             'You have object.env.request variable with full '
+             'webhook request.',
+        default='# You can use object.env.request variable '
+                'to get full data of webhook request.\n'
+                '# Example:\n#request.httprequest.'
+                'headers.get("X-Github-Event")',
+    )
+    python_code_get_ip = fields.Text(
+        'Get IP',
+        required=True,
+        help='Python code to get remote IP address '
+             'from request data.\n'
+             'You have object.env.request variable with full '
+             'webhook request.',
+        default='# You can use object.env.request variable '
+                'to get full data of webhook request.\n'
+                '# Example:\n'
+                '#object.env.request.httprequest.remote_addr'
+                '\nrequest.httprequest.remote_addr',
+
+    )
+    active = fields.Boolean(default=True)
+
+    @api.multi
+    def process_python_code(self, python_code, request=None):
+        """
+        Execute a python code with eval.
+        :param string python_code: Python code to process
+        :param object request: Request object with data of json
+                               and http request
+        :return: Result of process python code.
+        """
+        self.ensure_one()
+        res = None
+        eval_dict = {
+            'user': self.env.user,
+            'object': self,
+            'request': request,
+            # copy context to prevent side-effects of eval
+            'context': dict(self.env.context),
+        }
+        try:
+            res = safe_eval(python_code, eval_dict)
+        except BaseException:
+            error = tools.ustr(traceback.format_exc())
+            _logger.debug(
+                'python_code "%s" with dict [%s] error [%s]',
+                python_code, eval_dict, error)
+        if isinstance(res, basestring):
+            res = tools.ustr(res)
+        return res
+
+    @api.model
+    def search_with_request(self, request):
+        """
+        Method to search of all webhook
+        and return only one that match with remote address
+        and range of address.
+        :param object request: Request object with data of json
+                               and http request
+        :return: object of webhook found or
+                 if not found then return False
+        """
+        for webhook in self.search([('active', '=', True)]):
```

Comment:
`[('active', '=', True)]` is the default domain when an `active` field exists, unless you add `active_test=False` in the context. Any reason to enforce this domain ?